dolphin/resource_manager/storages.py

```diff
@@ -52,3 +53,46 @@ def get_all(req):
         if len(storage_all) == 0:
             break
     return build_storages(storage_all)
+
+
+def register(self, req, body):
+    """
+    This function for registering the new storage device
+    :param req:
+    :param body: "It contains the all input parameters"
+    :return:
+    """
+    # validate the body has all required parameters
+    required_parameters = ('hostname', 'manufacturer', 'model', 'username', 'password')
+
+    storage = body['storages']
+
+    for parameter in required_parameters:
+        if parameter not in storage:
+            msg = "Required parameter %s not found" % parameter
+            raise exc.HTTPUnprocessableEntity(explanation=msg)
+        if not storage.get(parameter):
+            msg = "Required parameter %s is empty" % parameter
+            raise exc.HTTPUnprocessableEntity(explanation=msg)
+
+    driver = drivermanager.DriverManager()
+    device_info = driver.register_storage(context, storage)
```

Comment:
what is context? we need to identify the role of context from Driver manager

Reply:
right. but since, register_storage accepting context as a parameter, i am passing this parameter.